tests/test_crypto.py

```diff
@@ -3884,6 +3949,38 @@ def test_get_verified_chain_invalid_chain_no_root(self):
         assert exc.value.args[0][2] == "unable to get issuer certificate"
         assert exc.value.certificate.get_subject().CN == "intermediate"
 
+    @pytest.fixture
+    def ca_file(self, tmpdir):
```

Comment:
I'm not particularly familiar with pytest's tmpdir fixture. Are these automatically cleaned up as part of the test suite?

Reply:
Yes, in a way. This fixture creates a unique temporary directory(*) for each test function invocation using it (so e.g. a new one for every parametric call), and performs a cleanup at the end of the pytest run. This cleanup keeps the temporary directories of the last 3 pytest runs, and deletes all others.
I admit I didn't give too much thought to this, as the `tmpdir` fixture is already used extensively in `test_ssl.py`, and indirectly via the `tmpfile` fixture defined in `conftest.py`.

(*) Technically the directory structure is like this: `pytest_base_temporary_directory/per_pytest_run_temporary_directory/per_test_invocation_temporary_directory`.)

Reply:
I had totally forgotten about the use in `test_ssl` because none of the maintainers of this project really like to think about it much 😂 . Thanks for the explanation anyway, this is fine.